bottomsheet-commons/src/main/java/com/flipboard/bottomsheet/commons/BottomSheetFragment.java

```diff
@@ -0,0 +1,257 @@
+        }
+    }
+
+    @Override
+    public LayoutInflater getLayoutInflater(Bundle savedInstanceState) {
+        if (!showsBottomSheet) {
+            return super.getLayoutInflater(savedInstanceState);
+        }
+        bottomSheetLayout = (BottomSheetLayout) findBottomSheetLayout();
+        if (bottomSheetLayout != null) {
+            return LayoutInflater.from(bottomSheetLayout.getContext());
+        }
+        return LayoutInflater.from(getContext());
+    }
+
+    @Nullable
+    private View findBottomSheetLayout() {
+        Fragment parentFragment = getParentFragment();
+        if (parentFragment != null) {
+            View view = parentFragment.getView();
+            if (view != null) {
+                return view.findViewById(bottomSheetLayoutId);
+            } else {
+                return null;
+            }
+        }
+        Activity parentActivity = getActivity();
+        if (parentActivity != null) {
+            return parentActivity.findViewById(bottomSheetLayoutId);
+        }
+        return null;
+    }
+
+    @Override
+    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
+
+        if (!showsBottomSheet) {
+            return;
+        }
+
+        View view = getView();
+        if (view != null) {
+            if (view.getParent() != null) {
+                throw new IllegalStateException("BottomSheetFragment can not be attached to a container view");
+            }
+            bottomSheetLayout.setOnSheetStateChangeListener(this);
+        }
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        if (bottomSheetLayout != null) {
+            viewDestroyed = false;
+            bottomSheetLayout.showWithSheetView(getView(), getViewTransformer(), this);
+        }
+    }
+
+    @Override
+    public void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        if (!showsBottomSheet) {
+            outState.putBoolean(SAVED_SHOWS_BOTTOM_SHEET, false);
+        }
+        if (backStackId != -1) {
+            outState.putInt(SAVED_BACK_STACK_ID, backStackId);
+        }
+        if (bottomSheetLayoutId != View.NO_ID) {
+            outState.putInt(SAVED_BOTTOM_SHEET_LAYOUT_ID, bottomSheetLayoutId);
+        }
+    }
+
+    /**
+     * Remove bottom sheet.
+     */
+    @Override
+    public void onDestroyView() {
+        super.onDestroyView();
+        if (bottomSheetLayout != null) {
+            viewDestroyed = true;
+            bottomSheetLayout.dismissSheet();
+            bottomSheetLayout = null;
+        }
+    }
+
+    @Override
+    @CallSuper
+    public void onDismissed(BottomSheetLayout bottomSheetLayout) {
+        if (!viewDestroyed) {
+            dismissInternal(true);
+        }
+    }
+
+    @Override
+    public void onSheetStateChanged(BottomSheetLayout.State state) {
```

Comment:
On second thought, I think it would be better not to set this. Could be confusing if the user set a listener in their activity and finds out that it's never called! They can still do the same thing in the fragment with `getBottomSheetLayout().setOnSheetStateChangeListener()`